middleware/pagination_test.go

```diff
@@ -0,0 +1,291 @@
+package middleware_test
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+
+	"github.com/redhatinsights/export-service-go/middleware"
+
+	chi "github.com/go-chi/chi/v5"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Handler", func() {
+	Describe("Test getLinks function", func() {
+		It("should return the proper Links struct", func() {
+			req, err := http.NewRequest("GET", "/test", nil)
+			Expect(err).To(BeNil())
+
+			data := make([]int, 100)
+
+			rr := httptest.NewRecorder()
+			applicationHandler := http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+				// (url *url.URL, p Paginate, data interface{})
+				links := middleware.GetLinks(r.URL, middleware.Paginate{Limit: 10, Offset: 20}, data)
```

Comment:
You might be able to simplify this a bit if you remove the handlerFunc "wiring" and test middleware.GetLinks() directly.